.github/workflows/test-unit.yml

```diff
@@ -26,19 +27,35 @@ jobs:
       - name: Setup line endings
         run: git config --global core.autocrlf false
 
+      - name: Change default TEST_COVERAGE when on main branch
+        if: github.ref == 'refs/heads/main'
+        run: echo "TEST_COVERAGE=true" >> $GITHUB_ENV
+
       - name: Checkout code
         uses: actions/checkout@v4
-      
+
       - name: Set TEST_KBC_PROJECTS
         uses: ./.github/actions/export-kbc-projects
         with:
           secrets: ${{ toJSON(secrets) }}
           projects: ${{ vars.TEST_KBC_PROJECTS }}
 
-      - name: Load Go cache
+      - name: Load Go build cache
         uses: ./.github/actions/go-cache
         with:
-          key: unit-${{ matrix.name }}
+          key: unit
+
+      - name: Change modtime of files to 1 unix timestamp
```

Comment:
Currently only `unit` tests are changing timestamps. E2e tests will be solved in another PR